app/assets/javascripts/index.js

```diff
@@ -255,17 +281,17 @@ $(document).ready(function () {
     var iframe = $('<iframe>')
         .hide()
         .appendTo('body')
-        .attr("src", url + querystring.stringify(query))
+        .attr("src", url() + querystring.stringify(query) + (android ? '#Intent;scheme=josm;end;':''))
```

Comment:
If `url` is going to stay as a function then I would just pass `query` in to it and let it figure out the whole URL in one go rather than splitting the logic for the intent URL across two different locations.